src/components/AudioInfo.vue

```diff
@@ -976,22 +799,22 @@ const checkRewritePreset = async () => {
 // 入力パラメータから、name以外のPresetを取得
 const presetPartsFromParameter = computed<Omit<Preset, "name">>(() => {
   if (
-    speedScaleSlider.state.currentValue.value == null ||
-    pitchScaleSlider.state.currentValue.value == null ||
-    intonationScaleSlider.state.currentValue.value == null ||
-    volumeScaleSlider.state.currentValue.value == null ||
-    prePhonemeLengthSlider.state.currentValue.value == null ||
-    postPhonemeLengthSlider.state.currentValue.value == null
+    parameters.value.some(
+      (parameter) => parameter.slider.state.currentValue.value == undefined
+    )
   )
     throw new Error("slider value is null");
 
   return {
-    speedScale: speedScaleSlider.state.currentValue.value,
-    pitchScale: pitchScaleSlider.state.currentValue.value,
-    intonationScale: intonationScaleSlider.state.currentValue.value,
-    volumeScale: volumeScaleSlider.state.currentValue.value,
-    prePhonemeLength: prePhonemeLengthSlider.state.currentValue.value,
-    postPhonemeLength: postPhonemeLengthSlider.state.currentValue.value,
+    ...parameters.value.reduce(
+      (acc, parameter) => ({
+        ...acc,
+        [parameter.key]: parameter.slider.state.currentValue.value,
+      }),
+      {} as {
+        [K in typeof parameters.value[number]["key"]]: number;
+      }
+    ),
```

Comment:
ここの型パズル、自分も挑戦してみたのですが難しいですね。。
今の形は`parameter.slider.state.currentValue.value`がnullだとしても通っちゃいそうです。（まあ真上でthrowしているので今のコードは大丈夫なのですが）

真上のnull時throwと混ぜてこんな感じにするのもありかなとか思いました。だいぶ読みづらいですが、まあより正しいはずではある･･･ 😇 
```suggestion
    ...parameters.value.reduce(
      (acc, parameter) => {
        if (parameter.slider.state.currentValue.value == undefined)
          throw new Error(
            `parameter(${parameter.key}).slider.state.currentValue.value == undefined`
          );
        acc[parameter.key] = parameter.slider.state.currentValue.value;
        return acc;
      },
      {} as {
        [K in typeof parameters.value[number]["key"]]: number;
      }
    ),
```

Reply:
真上でthrowしてるので、わざわざ書く必要もないかなぁって思ってます

Reply:
真上でthrowしてるのを下に統合しちゃうと型サポートも入って良い感じって意図でした。
TypeScriptの型サポートが中途半端なんすよね･･･